### PR TITLE
Make Streaming live logs work with multiple infrastructures

### PR DESCRIPTION
**Pull request by iszulcdeepsense**

### CODE REVIEW

File: lifecycle/lifecycle/server/cache.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+class LifecycleCache:
+    infrastructure_targets: dict[str, InfrastructureTarget] = {}
+
+    @classmethod
+    def on_plugins_reload(cls, plugin_engine: PluginEngine):
+        cls.infrastructure_targets = {}
+        for result in plugin_engine.invoke_plugin_hook(PluginCore.infrastructure_targets):
+            if result:
+                cls.infrastructure_targets.update(result)
```
[the remark this answers is not in the document]
Comment (iszulcdeepsense) [reply lifecycle/lifecycle/server/cache.py:14]: For performance. Whenever I needed infrastructure targets list, I called every plugin's hook to populate it. Instead, I could do it when the plugins are updated and keep the latest result.

[the remark this answers is not in the document]
Comment (Koshmaar) [reply lifecycle/lifecycle/server/cache.py:14]: Ok, makes sense.

File: lifecycle/lifecycle/server/socketio.py [hunk 2]
```diff
@@ -1,76 +1,124 @@
...
 
         self.wsgi_app = socketio.WSGIApp(self.sio, socketio_path='lifecycle/socket.io')
 
+    def open_logs_session(self, client_id: str, resource_properties: dict[str, str]) -> str:
+        logger.info(f'Creating log session for client: {client_id}')
+        job_name = resource_properties['job_name']
+        job_version = resource_properties['job_version']
+        tail = resource_properties.get('tail')
+        job = self.job_retriever.get_job(job_name, job_version)
+        job_version = job.version  # resolve version alias
```
[the remark this answers is not in the document]
Comment (Koshmaar) [reply lifecycle/lifecycle/server/socketio.py:82]: Ok, then maybe it should look like `job_version =  self.job_retriever.get_job(job_name, job_version).version  # resolve version alias` to have the comment on proper line. OTOH, it's longer one liner. Perhaps better option is to change comment to sth like `job_version = job.version  # contains resolved (non-aliased) version`. Your call.

File: lifecycle/lifecycle/server/socketio.py [hunk 3]
```diff
@@ -1,76 +1,124 @@
...
 logger = get_logger(__name__)
 
 
+class LogSessionDetails(BaseModel, arbitrary_types_allowed=True):
+    client_id: str
+    job_name: str
+    job_version: str
+    session_id: str
+    logs_streamer: LogsStreamer
+    tail: str | None
```
[the remark this answers is not in the document]
Comment (iszulcdeepsense) [reply lifecycle/lifecycle/server/socketio.py:26]: It should be int, just changed it to:
```
tail: int | None  # number of recent lines to show
```

File: lifecycle/lifecycle/server/socketio.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (iszulcdeepsense) [reply lifecycle/lifecycle/server/socketio.py:82]: But it's not the only reason of getting the job details. In addition, I take infrastructure_target name from it as well.

[the remark this answers is not in the document]
Comment (iszulcdeepsense) [reply lifecycle/lifecycle/server/socketio.py:82]: Surely, I can change the comment, if you feel it would be more understandable.

**Pull request merged**